addons/test_mail/tests/test_message_post.py

```diff
@@ -823,6 +823,53 @@ def test_message_post_w_attachments(self):
                         ]
         )
 
+    @mute_logger('odoo.addons.mail.models.mail_mail', 'odoo.addons.mail.models.mail_message_schedule')
+    def test_message_post_filter_main_attachments(self):
+        """
+        Test the message_main_attachment heuristics with an emphasis on the XML/PDF types.
+        -> we don't want XML files to be set as message_main_attachment
+        """
+        _attachments = [
+            ('List1', b'My xml attachment'),
+            ('List2', b'My pdf attachment'),
+        ]
+        _attachment_records = self.env['ir.attachment'].create(
+            self._generate_attachments_data(3, 'mail.compose.message', 0)
+        )
+        _attachment_records[0].write({'mimetype': 'application/xml'})
+        _attachment_records[1].write({'mimetype': 'application/pdf'})
+
+        test_record = self.env['mail.test.simple'].browse(self.test_record.ids)
+        self.assertFalse(test_record.message_main_attachment_id)
+
+        with self.mock_mail_gateway():
+            test_record.message_post(
+                attachments=_attachments[0],
```

Comment:
Would use meaningful variables, like 'xml_attachment_data, xml_attachment' and 'pdf_attachment_data, pdf_attachment'. That way we better understand the subtests (otherwise have to go back and deduce that 0 are xml, 1 are pdf).